internal/services/gateway/action/run.go

```diff
@@ -429,10 +430,12 @@ func (h *ActionHandler) CreateRuns(ctx context.Context, req *CreateRunRequest) e
 
 	var variables map[string]string
 	if req.RunType == itypes.RunTypeProject {
-		var err error
-		variables, err = h.genRunVariables(ctx, req)
-		if err != nil {
-			return err
+		if req.RefType != itypes.RunRefTypePullRequest || req.PRFromSameRepo || (!req.PRFromSameRepo && req.Project.PassVarsToForkedPR) {
```

Comment:
We could also simplify this to:
if req.RefType != itypes.RunRefTypePullRequest || req.PRFromSameRepo || req.Project.PassVarsToForkedPR